nlp/twitter.py

```diff
@@ -118,32 +46,205 @@ def cleanTweet(tweet):
+
+
+    def open_stream(self):
+        """PART 3: USE CREDENTIALS TO OPEN STREAM"""
+        # I think we can only open one stream at a time
+        QUERY = '&'.join(EMOJIS[0])
+
+        stream = self.MyStreamer(
+            self.TWITTER_KEY,
+            self.TWITTER_SECRET,
+            self.OAUTH_TOKEN,
+            self.OAUTH_TOKEN_SECRET
+        )
+
+        stream.statuses.filter(track=QUERY, language='en')
+
+
+    def connect_twitter_stream(self):
+        """Connects to Twitter Stream API."""
+        # set up authentication using oauth v1.0
+        self.twitter_auth()
+        # open stream (tweets in real-time)
+        self.open_stream()
+
+
+    def connect_twitter_search(self):
+        """Connects to Twitter Search API.
+
+        Simpler as it doesn't require oauth.
+        NOTE: tweets are limited to 100 at a time.
+        """
+        twitter = Twython(
+            self.TWITTER_KEY,
+            self.TWITTER_SECRET,
+            oauth_version=2
+        )
+        TWITTER_ACCESS_TOKEN = twitter.obtain_access_token()
+
+        twitter = Twython(self.TWITTER_KEY, access_token=TWITTER_ACCESS_TOKEN)
+
+        for emoji in EMOJIS:
+            QUERY = '&'.join(emoji)
+            RESULTS = twitter.search(
+                q=QUERY,
+                count=100,
+                result_type='mixed',
+                lang='en'
+            )
+
+            for tweet in RESULTS['statuses']:
+                clean_tweet(tweet['text'])
+
+
+    def __init__(self, use_search):
+        """Load pickle file if it exists and open stream or search."""
         global emoji_dict
-        emoji_dict = pickle.load(inputFile)
-        inputFile.close()
 
-    # Choose whether to use the stream or search Twitter API
-    connectTwitterStream()
-    # connectTwitterSearch()
+        signal.signal(signal.SIGINT, self.sig_int_handler)
 
-def keyboardInterruptHandler(signal, frame):
-    global emoji_dict
-    print('KeyboardInterrupt'.format(signal))
+        if (Path('nlp/tweets')).is_file():
+            filename = 'nlp/tweets'
+        else:
+            filename = 'tweets'
+
+        if os.path.getsize(filename) > 0:
+            in_file = open(filename, 'rb')
+            emoji_dict = pickle.load(in_file)
+            in_file.close()
+
+        try:
+            if use_search:
+                self.connect_twitter_search()
+            else:
+                self.connect_twitter_stream()
+
+        except (
+            exceptions.TwythonAuthError,
+            exceptions.TwythonError,
+            ValueError
+        ) as error:
+            print(error)
+            print('Missing TWITTER_KEY and TWITTER_SECRET env vars')
+
+
+    def sig_int_handler(self, sig, frame):
+        """Handles SIGINT, writing to pickle with all retrieved data."""
+        global emoji_dict
+
+        print(f'Terminated by {signal.SIGINT.name}.')
+
+        filename = 'tweets' if 'nlp' in os.getcwd() else 'nlp/tweets'
```

Comment:
oh did you ever get the search tweets to be added to the pickle? i think it's because it's only in the sig int handler. need to add these lines to the end of connect_twitter_search() too. i realize now i added all the file writing stuff after testing the search. i can add it in the next PR.

Reply:
ah, no worries. thanks for the heads up about this. i totally missed that, and that's why the search wasn't yielding any updates to the pickle file. we're all good now. feel free to rearrange stuff or do whatever to make things easier for you.